src/org/aavso/tools/vstar/ui/dialog/period/PeriodAnalysis2DResultDialog.java

```diff
@@ -219,29 +238,57 @@ private JTabbedPane createTabs() {
 	protected void newPhasePlotButtonAction() {
 		PeriodChangeMessage message = new PeriodChangeMessage(this,
 				selectedDataPoint.getPeriod());
+		message.setName(this.getName());
 		Mediator.getInstance().getPeriodChangeNotifier()
 				.notifyListeners(message);
 	}
 
 	@Override
 	protected void findHarmonicsButtonAction() {
-		List<Double> data = algorithm.getResultSeries().get(
-				PeriodAnalysisCoordinateType.FREQUENCY);
-		List<Harmonic> harmonics = findHarmonics(
-				selectedDataPoint.getFrequency(), data);
+		String chartID = null;
+		Component c = tabbedPane.getSelectedComponent();
+		if (c instanceof PeriodAnalysis2DChartPane) {
+			chartID = ((PeriodAnalysis2DChartPane)c).getChartPaneID();
+		}
+		
+		if (chartID == null) {
+			MessageBox.showMessageDialog("Find Harmonic", "Please select a tab with a chart");
```

Comment:
I am wondering about this. Why are we not permitting the top-hits (or data) table to be selected? It seems quite intuitive to me for that to be permitted in addition to a plot tab, especially since we are now limiting ourselves to searching top-hits (which is fine) for harmonics.